contracts/Staking.sol

```diff
@@ -201,27 +192,43 @@ contract Staking is IStaking, Initializable, OwnableUpgradeable, Constants {
         }
     }
 
-    function _checkDelegateLimitation(address _indexer, uint256 _amount) private view {
+    function _checkDelegateLimitation(address _indexer, uint256 _amount) external onlyStakingManager {
         require(
             delegation[_indexer][_indexer].valueAfter * indexerLeverageLimit >=
                 totalStakingAmount[_indexer].valueAfter + _amount,
             'S002'
         );
     }
 
+    function _registerIndexer(address _indexer) external onlyStakingManager {
```

Comment:
use `addIndexer` and `removeIndexer` as it only change the indexers related states but not include the full `indexerRegistry` features